django/db/models/query.py

```diff
@@ -927,6 +927,10 @@ def get_or_create(self, defaults=None, **kwargs):
         Return a tuple of (object, created), where created is a boolean
         specifying whether an object was created.
         """
+        if self._known_related_objects:
+            field, related_objects = next(iter(self._known_related_objects.items()))
+            rel_obj = next(iter(related_objects.values()))
+            kwargs[field.name] = rel_obj
```

Comment:
I would suggest extracting this to a method as it is duplicated, e.g. `self._possibly_inject_fk_val(kwargs)`

And then you can add a docstring to that explaining what this is all about.

Reply:
Thank you! It looks much cleaner.